src/components/Square.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useContext, useEffect } from 'react'
+import { GameStateContext } from './Game'
+import { updateBoardState } from 'utils'
+
+//Square props
+type SquareProps = {
+  id: number
+  isXNext: boolean
+  squareLetter: string
+  disabled?: boolean
+  isHighlighted: boolean
+}
+
+//component
+const Square = ({ id, isXNext, disabled, isHighlighted }: SquareProps) => {
+  const { gameState, stateDispatch } = useContext(GameStateContext)
+
+  //take care updating the game history update after render
+  useEffect(() => {
+    stateDispatch({ type: 'SET_NEXT', newGameState: gameState })
```

Comment:
definitely wouldn't put these here, random re-renders could cause some weird side effects and you probably end up calling it excessively. Seems like you'd want this more closely tied to your even handlers for resolving a turn